azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/ConfigMojo.java

```diff
@@ -322,82 +369,78 @@ private WebAppConfiguration getRuntimeConfiguration(WebAppConfiguration configur
     }
 
     private WebAppConfiguration.Builder getRuntimeConfigurationOfLinux(WebAppConfiguration.Builder builder,
-                                                                       WebAppConfiguration configuration)
-        throws MojoFailureException {
-        final String defaultJavaVersion = configuration.getLinuxJavaVersionOrDefault();
-        final String javaVersion = queryer.assureInputFromUser("javaVersion",
-            defaultJavaVersion, RuntimeStackUtils.getValidJavaVersions(), null);
+            WebAppConfiguration configuration) throws MojoFailureException {
+        String defaultJavaVersion = configuration.getLinuxJavaVersionOrDefault();
+        // sometimes the combination of <javaVersion> and <webContainer> may not be valid, but
+        // <javaVersion> might be right
+        if (Objects.isNull(configuration.getRuntimeStack()) && Objects.nonNull(this.getRuntime()) &&
+                Objects.nonNull(this.getRuntime().getJavaVersion())) {
+            defaultJavaVersion = JavaVersionUtils.formatJavaVersion(this.getRuntime().getJavaVersion());
+        }
+        final String javaVersion = queryer.assureInputFromUser(JAVA_VERSION, defaultJavaVersion,
+                JavaVersionUtils.getValidJavaVersions(), String.format(COMMON_PROMPT, JAVA_VERSION, defaultJavaVersion));
         // For project which package is jar, use java se runtime
         if (isJarProject()) {
-            return builder.runtimeStack(RuntimeStackUtils.getRuntimeStack(javaVersion));
+            return builder.runtimeStack(RuntimeStackUtils.getJavaSERuntimeStack(javaVersion));
+        }
+        final List<String> validRuntimeStacks = RuntimeStackUtils.getValidLinuxRuntimeStacksForJavaVersion(javaVersion);
+        String defaultLinuxRuntimeStack = configuration.getLinuxRuntimeStackOrDefault();
+        // sometimes the combination of <javaVersion> and <webContainer> may not be valid, but
+        // <webContainer> might be right
+        if (Objects.isNull(configuration.getRuntimeStack()) && Objects.nonNull(this.getRuntime()) &&
```

Comment:
It may be a bit confused to show validation result here, we may show validation before `config` and only use correct default value here

Reply:
If we don't show it here, user will be strange why his values are not replaced(if his value is illegal).